src/main/scala/ignition/core/http/AsyncHttpClientStreamApi.scala

```diff
@@ -1,17 +1,35 @@
 package ignition.core.http
 
 import java.io.InputStream
+import java.net.{URL, URLDecoder, URLEncoder}
 import java.util.concurrent.TimeUnit
 
 import akka.util.Timeout
+import spray.http.Uri.{Path, Query}
 import spray.http._
 
 import scala.concurrent.Future
 import scala.concurrent.duration._
 import scala.language.postfixOps
 
-
 object AsyncHttpClientStreamApi {
+
+  private def sanitizePathSegment(segment: String) =
+    URLEncoder.encode(URLDecoder.decode(segment, "UTF-8"), "UTF-8").replace("+", "%20")
```

Comment:
http://stackoverflow.com/questions/1634271/url-encoding-the-space-character-or-20